src/main/java/com/snowflake/kafka/connector/SnowflakeSinkConnectorConfig.java

```diff
@@ -129,6 +134,13 @@ public class SnowflakeSinkConnectorConfig {
   public static final String SNOWPIPE_STREAMING_MAX_CLIENT_LAG =
       "snowflake.streaming.max.client.lag";
 
+  public static final String SNOWPIPE_STREAMING_MAX_CHANNEL_SIZE =
+      "snowflake.streaming.max.channel.size";
+  public static final long SNOWPIPE_STREAMING_MAX_CHANNEL_SIZE_DEFAULT = 128 * 1024 * 1024;
+
+  public static final String SNOWPIPE_STREAMING_MAX_MEMORY_LIMIT =
+      "snowflake.streaming.max.memory.limit";
+  public static final long SNOWPIPE_STREAMING_MAX_MEMORY_LIMIT_DEFAULT = -1L;
```

Comment:
What's the reason of adding these parameters given that we already have SNOWPIPE_STREAMING_CLIENT_PROVIDER_OVERRIDE_MAP that can control these parameters?

Reply:
We want to explicitly define them for better visibility and put emphasis on them so the customer is aware that they exist and can be modified. The override map was kinda hidden and only used in rare occasions as stated in the description "optional and recommended to use ONLY after consulting Snowflake Support"

Reply:
Yes, that's why I was asking for a design doc to describe the overall plan :) It seems to me that the decision is that we're exposing more parameters for customer to control which is not ideal in terms of ease of use, now they need to understand and use the new parameters instead of the old parameters on `buffer.*`. For example, I was imagine that the buffer size can still be controlled by `buffer.size.bytes` and we can use that value to set the max channel size so that there is no behavior change. 

Another concern is exposing the MAX_MEMORY_LIMIT, I'm not sure customer would be able to understand this easily and use it in a correct way since the number of clients and channels are controlled by Kafka Connector. I don't think we have heard from customers in the past that this needs to be updated.

Reply:
+1 with @sfc-gh-tzhang 
Please lets add new configs only if it is absolutely needed. 
A design doc describing what this will do would be very helpful.